Parallelize the Renderer

pfent opened the pull request:
Sprinkle some `#pragma omp`s all over the Renderer.

Rendering times for my 471 pages test document with 4 cores:
Before: ~25s. After: ~7s (approx. x3.5 speedup)

pfent committed:
parallelize the Renderer with openmp

pfent committed:
Update README.md

neumannt commented on Renderer.cpp:
```diff
@@ -93,7 +93,12 @@ void Renderer::run()
-   for (unsigned index=0;(index<images.size())&&(!mustStop);index++) {
+#pragma omp parallel for schedule(dynamic)
+   for (unsigned index=0;index<images.size();index++) {
+       if(mustStop) { // break
```
please add a space after the if

neumannt commented on Renderer.cpp:
```diff
@@ -93,7 +93,12 @@ void Renderer::run()
+#pragma omp parallel for schedule(dynamic)
+   for (unsigned index=0;index<images.size();index++) {
+       if(mustStop) { // break
+           index = std::numeric_limits<unsigned>::max();
```
this will result in an endless loop: you set index to max, but increase it afterwards (after the continue). Better set it to images.size()

pfent committed:
matching formatting with existsting code

pfent committed:
break the loop by with index = images.size()
this avoids an overflow with potential for infinite loops since
numeric_limits::max() + 1 == 0, which always fulfills the condition
